membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -198,16 +196,22 @@ class AccountController(
 
         logger.info(s"Attempting to retrieve payment details for identity user: $userId")
 
-        paymentDetails(userId, filter, request.touchpoint).toEither
-          .map {
-            case Right(subscriptionList) =>
-              logger.info(s"Successfully retrieved payment details result for identity user: $userId")
-              val response = ProductsResponse.from(user, subscriptionList)
-              Ok(Json.toJson(response))
-            case Left(message) =>
-              logger.warn(s"Unable to retrieve payment details result for identity user $userId due to $message")
-              InternalServerError("Failed to retrieve payment details due to an internal error")
-          }
+        for {
+          catalog <- request.touchpoint.futureCatalog
+          result <- paymentDetails(userId, filter, request.touchpoint).toEither
+            .map {
+              case Right(subscriptionList) =>
+                logger.info(s"Successfully retrieved payment details result for identity user: $userId")
+                val productsResponseWrites = new ProductsResponseWrites(catalog)
```

Comment:
toJson's implicit writes (used just below) now need the catalog, so I've put the code in a class `ProductsResponseWrites` and passed the catalog in.